src/backend/access/heap/heapam.c

```diff
@@ -400,7 +400,7 @@ heapgetpage(TableScanDesc sscan, BlockNumber page)
 	CHECK_FOR_INTERRUPTS();
 
 	/* Prefetch next block */
-	if (enable_seqscan_prefetch)
+	if (enable_seqscan_prefetch && !smgr_prefetch_in_progress(scan->rs_base.rs_rd->rd_smgr))
```

Comment:
No, it is obviously not only fo prewarming.
It is for any seqscan prefetch.
How perfetch was implemented implemented (before this patch)?

Before each `ReadBufferextended` in `heapgetpage` there is prefetch loop:
```
	if (enable_seqscan_prefetch)
	{
              ...
              	for (int i = 1; i <= prefetch_limit; i++)
              	{
              	     PrefetchBuffer(scan->rs_base.rs_rd, MAIN_FORKNUM, page+i);
              	}
        }
	/* read page using selected strategy */
	scan->rs_cbuf = ReadBufferExtended(scan->rs_base.rs_rd, MAIN_FORKNUM, page,
									   RBM_NORMAL, scan->rs_strategy);
```

So when we request page with address A, we initiate prefetch of pages A+1, A+2, ...,A+N 
where N=seqscan_prefetch_buffers.
Then the following things can happen in ReadBuffer:
1. Requested page is already present in shared buffers. In this case smgr is not accessed at all. We need to somehow remove all prefetch requests.
2. Page is not present in shared buffers and smgread is called. Then there are once agai two choices:
    2.1 There are some prefetch requests in progress (sent to pageserver). In this case we should wait for responses from pageserver. If response with this address A is found then we are done (prefetch hit). Otherwise we have to consume all prefetch responses and got to 2.2.
    2.2 We have to request page A  fro page server. We combine this request with N prefetch requests and send them together to pageserver.
    
So, except cases 2.2 prefetch requests  are not used and has to be ignored.
As far as each PrefetchBuffer first check presence of the  requested page in shared buffers, it adds quite esential overhead. Especially in case 1 - when all pages are present in memory. In this case we do N+1 buffer hsh lokups instead of  just one.   It increase query execution time 3 times.   

So this PR is attempt to address this problem. Cn it be done in different way? Obviously!
But I want to make our prefetch compatible with Postgres prefetch mechanism and that is why it is using PrefetchBuffer. To address the problem above I have to extent prefetch API by adding prefetch_in_progress() method.